### ISSUE TITLE
Fix double heading conversion for ego_agent_past during data augmentation smoothing

### ISSUE THREAD

**Issue by Max-Bin** [tier4/Diffusion-Planner#185]
## Problem

`ego_agent_past` is converted to 4D `[x, y, cos, sin]` before data augmentation in `train_epoch.py`, but `StatePerturbation.centric_transform()` still treats `ego_agent_past[..., 2]` as a raw heading angle and applies `torch.cos()` / `torch.sin()` again.

This double conversion corrupts the ego-history heading used by `smoothing_future_trajectory()`, and the smoothed result is written back as the ego future training target.

## Evidence

In `diffusion_planner/diffusion_planner/train_epoch.py`, `ego_agent_past` is normalized to 4D before augmentation:

```python
inputs["ego_agent_past"] = heading_to_cos_sin(inputs["ego_agent_past"])
```

`heading_to_cos_sin()` is already idempotent for 4D input, so after this point the layout is `[x, y, cos(theta), sin(theta)]`.

However, in `diffusion_planner/diffusion_planner/utils/data_augmentation.py`, `StatePerturbation.centric_transform()` rebuilds `ego_past4d` as:

```python
ego_past4d = torch.cat(
    [
        inputs["ego_agent_past"][..., :2],
        torch.cos(inputs["ego_agent_past"][..., 2:3]),
        torch.sin(inputs["ego_agent_past"][..., 2:3]),
    ],
    dim=-1,
)
```

At this point `inputs["ego_agent_past"][..., 2]` is already `cos(theta)`, not `theta`. The reconstructed heading therefore becomes:

```text
atan2(sin(cos(theta)), cos(cos(theta))) = cos(theta)
```

For example, ego-centric heading `theta = 0` becomes `1 rad` instead of `0 rad`.

## Impact

The corrupted `ego_past4d` is passed into `smoothing_future_trajectory()`:

```python
ego_future4d = smoothing_future_trajectory(
    ego_past4d, inputs["ego_current_state"], ego_future4d
)
```

The smoothed future is then converted back and assigned as the ego future training target:

```python
inputs["ego_agent_future"] = ego_future
```

This can directly pollute the ego future target used for training when data augmentation and smoothing are enabled.

Default training settings enable this path:

- `use_data_augment=True`
- `use_smoothing_future_trajectory=True`

`augment_prob` only controls current-state perturbation, but `centric_transform()` and smoothing are still applied through the augmentation path when augmentation is enabled.

## Proposed fix

Use the already-4D ego history directly when the last dimension is 4:

```python
ego_past = inputs["ego_agent_past"]
if ego_past.shape[-1] == 4:
    ego_past4d = ego_past
else:
    ego_past4d = torch.cat(
        [
            ego_past[..., :2],
            torch.cos(ego_past[..., 2:3]),
            torch.sin(ego_past[..., 2:3]),
        ],
        dim=-1,
    )
```

The minimal one-line fix for the current training path is:

```python
ego_past4d = inputs["ego_agent_past"]
```

## Validation suggestion

Add a regression test where `ego_agent_past` is already `[x, y, cos, sin]` with zero heading and verify that the history heading passed to smoothing remains zero instead of becoming `1 rad`.
